deephyper/core/analytics/_db_manager.py

```diff
@@ -0,0 +1,127 @@
+import json
+import os
+import abc
+import getpass
+import subprocess
+
+import pandas as pd
+import yaml
+from datetime import datetime
+from tinydb import TinyDB, Query
+
+class DBManager(abc.ABC):
+    """Database Manager, for saving DeepHyper experiments and accessing/modifying the resulting databases.
+
+    Example Usage:
+
+        >>> dbm = DBManager(user_name="Bob", path="path/to/db.json")
+
+    Args:
+        user_name (str, optional): the name of the user accessing the database. Defaults to ``os.getlogin()``.
+        path (str, optional): the path to the database. Defaults to ``"~/.deephyper/local_db.json"``.
+    """    
+    def __init__(self, user_name: str = getpass.getuser(), path: str = "~/.deephyper/local_db.json") -> None:
+        self._user_name = user_name
+        self._db = TinyDB(path)
+
+    def _get_pip_env(self, pip_versions=None):
+        if pip_versions is None:
+            pip_list_com = subprocess.run(['pip', 'list', '--format', 'json'], stdout=subprocess.PIPE)
+            pip_env = json.loads(pip_list_com.stdout.decode('utf-8'))
+        else:
+            with open(pip_versions, 'r') as file:
+                pip_env = json.load(file)
+        return pip_env
+    
+    def add(self, log_dir: str, label: str = None, description: str = None, pip_versions: str = None, extra_metadata: dict = None):
+        """Adds an experiment to the database.
+
+        Example Usage:
+            
+            >>> extra = {"machine": "ThetaGPU", "n_nodes": 4, "num_gpus_per_node": 8}
+            >>> dbm.add("path/to/search/log_dir/", label="exp_101", description="The experiment 101", exra_metadata=extra)
+
+        Args:
+            log_dir (str): the path to the search's logging directory.
+            label (str, optional): the label wished for the experiment. Defaults to None.
+            description (str, optional): the description wished for the experiment. Defaults to None.
+            pip_versions (str, optional): the path to the ``.json`` file corresponding to the ouptut of ``pip list --format json``. Defaults to None.
```

Comment:
`pip_versions` should be
* `True` by default (we run the pip version checkpointing)
* `False` we don't want to store any pip version checkpoint
* `path/to/pip_version.json` in the case where the user already has it